pytorch_lightning/trainer/training_tricks.py

```diff
@@ -40,7 +40,7 @@ def clip_gradients(self):
                 device = parameters[0].device
                 total_norm = torch.zeros([], device=device if parameters else None)
                 for p in parameters:
-                    param_norm = p.grad.data.norm(norm_type) ** norm_type
+                    param_norm = p.grad.data.pow(norm_type).sum()
```

Comment:
```suggestion
                    param_norm = p.grad.data.abs().pow(norm_type).sum()
```
now that is what I call a norm :)

Reply:
the other one is general to not just the 1 or 2 norm no?

Reply:
yes for all odd integers for example. Definition of p-norm: https://en.wikipedia.org/wiki/Lp_space#The_p-norm_in_finite_dimensions

Reply:
right, sorry I thought we were modifying the implementation for 
```Trainer(track_grad_norm)```
and so i was wondering how we would generalize to the p norm because your suggestion isn't general to the p-norm

Reply:
my suggestion is not? which case does it not cover? the inf case is covered in the if clause above.

Reply:
oh, just saw that the norm_type is hardcoded as `norm_type = float(2.0)`. That is very strange.

Reply:
so it was merged without the `abs(...)` ? @awaelchli @williamFalcon

Reply:
> oh, just saw that the norm_type is hardcoded as `norm_type = float(2.0)`. That is very strange.

Right, this is why I omitted the .abs() call as the 2-norm is hardcoded here. However, I would also favor your more general suggestions since this would be safer if the `norm_type` will be made variable in the future.